api/cloud/oc_cloud_resource.c

```diff
@@ -161,13 +145,13 @@ post_cloud(oc_request_t *request, oc_interface_mask_t interface,
   bool request_invalid_in_state = true;
   switch (ctx->store.cps) {
   case OC_CPS_UNINITIALIZED:
-  case OC_CPS_READYTOREGISTER:
-  case OC_CPS_FAILED:
-  case OC_CPS_DEREGISTERING:
     request_invalid_in_state = false;
     break;
+  case OC_CPS_READYTOREGISTER:
```

Comment:
@jkralik I think this might be the opposite of what we wanted, certainly the hub tests don't like it.

Reply:
why ? if we moved it to line 145 it will be fixed ?

Reply:
After this change in `OC_CPS_READYTOREGISTER`, `OC_CPS_DEREGISTERING`, and  `OC_CPS_FAILED` state you can only set empty "cis", for every other update the `request_invalid_in_state` will be true and thus line 165 will return OC_STATUS_BAD_REQUEST. Previously, in these states you could do all kinds of updates. It seems that at least the hub tests require the device to be configurable in the `OC_CPS_READYTOREGISTER` state. Or do we want to rewrite tests in plgd repos?

Reply:
Oh ok. For these you can allow too configure it with all 4 values.